### PR TITLE
Fix #992: carma-messenger UI - Truck picture not showing consistently

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
The truck image display a flashing broken link when invalid VIN number is received

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/992

## Motivation and Context
ACE testing

## How Has This Been Tested?
Local integration testing 

## Types of changes
JS logic change

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by dan-du-car** [comment]
Update the carma.config.js parameters with the list of carma-messenger valid VIN numbers.  This VIN numbers should match the supported truck images.

File: carma-messenger-ui/website/widgets/truckInspection/widget.js [hunk 1]
```diff
@@ -313,6 +318,22 @@ CarmaJS.WidgetFramework.truckInspection = (function () {
...
+    {
+        let isValid = false;
+        let valid_truck_vin_numbers = CarmaJS.Config.getValidTruckVINNumber();
+        valid_truck_vin_numbers.forEach((valid_vin)=>{
+            console.log("Valid VIN: " + valid_vin);
+            console.log("Check VIN: " + vin_number);
+            if(valid_vin.trim() == vin_number.trim()){
+                console.log("Found a valid VIN");
+                isValid = true;
+            }
```
Comment (maefromm) [inline carma-messenger-ui/website/widgets/truckInspection/widget.js:332]: @dan-du-car please put an else statement to show "invalid vin found"

**Review by maefromm — CHANGES_REQUESTED** [review]

File: carma-messenger-ui/website/widgets/truckInspection/widget.js [hunk 2]
```diff
@@ -313,6 +318,22 @@ CarmaJS.WidgetFramework.truckInspection = (function () {
...
             console.log("truckInspectionSafetyLogPerVIN element is created");
     }
+
+    var IsValidVINNumber = function(vin_number)
+    {
+        let isValid = false;
+        console.log("Check VIN: " + vin_number);
+        let valid_truck_vin_numbers = CarmaJS.Config.getValidTruckVINNumber();
+        valid_truck_vin_numbers.forEach((valid_vin)=>{
+            console.log("Valid VIN: " + valid_vin);
```
Comment (maefromm) [inline carma-messenger-ui/website/widgets/truckInspection/widget.js:328]: This is not valid VIN yet. So don't think you need this extra log. Or just don't say it's Valid yet.

File: carma-messenger-ui/website/widgets/truckInspection/widget.js [hunk 3]
```diff
@@ -313,6 +318,22 @@ CarmaJS.WidgetFramework.truckInspection = (function () {
...
+            if(valid_vin.trim() == vin_number.trim()){
```
Comment (maefromm) [inline carma-messenger-ui/website/widgets/truckInspection/widget.js:329]: If you've trimmed on line 23, i think what you want to check is the actual valid_vin "as is", without having to trim it again to compare "as is" the "value" you have stored and using for the jpg name.

File: carma-messenger-ui/website/widgets/truckInspection/widget.js (hunk 1 above)
Comment (maefromm) [reply carma-messenger-ui/website/widgets/truckInspection/widget.js:332]: I would still prefer to have console log saying: "Found valid VIN #####" or Found invalid VIN #####. especially if there's only one item we're iterating in the forEach((valid_vin)) loop.

**Review by maefromm — CHANGES_REQUESTED** [review]

File: carma-messenger-ui/website/widgets/truckInspection/widget.js (hunk 3 above)
Comment (dan-du-car) [reply carma-messenger-ui/website/widgets/truckInspection/widget.js]: Addressed

File: carma-messenger-ui/website/widgets/truckInspection/widget.js (hunk 2 above)
Comment (dan-du-car) [reply carma-messenger-ui/website/widgets/truckInspection/widget.js]: Addressed

**Review by maefromm — APPROVED** [review]

**Review by maefromm — APPROVED** [review]

**Pull request merged**